chia/full_node/full_node.py

```diff
@@ -1294,6 +1294,7 @@ def has_valid_pool_sig(self, block: Union[UnfinishedBlock, FullBlock]) -> bool:
             and block.foliage.prev_block_hash != self.constants.GENESIS_CHALLENGE
             and block.reward_chain_block.proof_of_space.pool_public_key is not None
         ):
+            assert block.foliage.foliage_block_data.pool_signature is not None
```

Comment:
apparently there's no test covering this line. I think it's trivially correct and warrants an exception. In the long run this function should probably be pulled out as a free function and unit tested